tests/ftest.rb

```diff
@@ -0,0 +1,133 @@
+  sleep 60
+end
+
+def prepare_es
+  puts('Wipe existing data')
+  system('curl -X DELETE http://localhost:9200/mongo --user elastic:changeme')
+end
+
+def set_auth
+  config_file = File.expand_path("#{File.dirname(__FILE__)}/../config/connectors.yml")
+
+  body = {
+    "name": 'mongo-connector',
+    "role_descriptors": {
+      "mongo-connector-role": {
+        "cluster": ['all'],
+        "index": [
+          {
+            "names": ['*'],
+            "privileges": ['all']
+          }
+        ]
+      }
+    }
+  }
+  response = es_client.perform_request('POST', '_security/api_key',
+                                       {}, body)
+
+  api_key = response.body['encoded']
+  # copy it into the config
+  config = YAML.load_file(config_file)
+  config['elasticsearch']['api_key'] = api_key.strip
+  File.open(config_file, 'w') { |file| file.write(config.to_yaml) }
+end
+
+def run_sync
+  # late import because the lib import the config when loaded
+  # and we change it in `set_auth`
+  $LOAD_PATH << LIB_DIR
+
+  Dir["#{LIB_DIR}/**/*.rb"].sort.each { |f|
+    next if f.include?('lib/app')
+    next if f.include?('connectors_async')
+
+    absolute_dir = File.dirname(f)
+    relative_dir = absolute_dir.sub(/.*lib\//, '')
+    name = File.basename(f, '.rb')
+    f = File.join(relative_dir, name)
+    require(f)
+  }
+
+  require 'active_support/json'
+  require 'core/elastic_connector_actions'
+  require 'core/connector_settings'
+
+  puts('Create the connector')
+  Core::ElasticConnectorActions.ensure_connectors_index_exists
+  connector_id = Core::ElasticConnectorActions.create_connector('mongo', 'mongo')
+
+  config = {
+    :host => {
+      :value => '127.0.0.1:27018'
+    },
+    :database => {
+      :value => 'sample_airbnb'
+    },
+    :collection => {
+      :value => 'listingsAndReviews'
+    }
+  }
+
+  Core::ElasticConnectorActions.update_connector_configuration(connector_id, config)
+  Core::ElasticConnectorActions.update_connector_status(connector_id, 'configured')
+  puts("Connector id #{connector_id}")
+  puts('Syncing')
+
+  config_settings = Core::ConnectorSettings.fetch(connector_id)
+
+  # Creating the mapping for the index
+  settings = Utility::Elasticsearch::Index::TextAnalysisSettings.new(:language_code => 'en', :analysis_icu => false).to_h
+  mappings = Utility::Elasticsearch::Index::Mappings.default_text_fields_mappings(:connectors_index => true)
+  body_payload = { settings: settings, mappings: mappings }
+  Core::ElasticConnectorActions.ensure_index_exists(config_settings[:index_name], body_payload)
+
+  Core::SyncJobRunner.new(config_settings, App::Config[:service_type]).execute
+
+  puts('Sync done.')
+end
+
+def verify
+  puts('Verifying data now')
+  count = es_client.count(index: 'mongo').body['count']
+  raise StandardError.new "Bad Count #{count}" unless count > 5000
+  puts('Verified!')
+end
+
+if __FILE__ == $PROGRAM_NAME
+  start_stack
+  prepare_es
+  set_auth
+  run_sync
+  verify
```

Comment:
There should be a cleanup step?

Reply:
good point! adding it -- I will simply teardown the stack